facility-all-programs.ipynb

```diff
@@ -461,32 +488,97 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## This section optionally saves some of this data to a CSV file in your Google Drive.\n",
-    "The first of the next two cells will open your Google Drive to allow the file to be written there.  Then you must select one of the data sets to write from the dropdown list.  The next cell writes the data to your drive.  You can return to the dropdown to write additional data files.\n"
+    "### Compile and chart information by program\n",
+    "Look at violations, inspections, and enforcements at this facility for each program - the CAA, CWA, and RCRA"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "scrolled": true
+   },
    "outputs": [],
    "source": [
-    "from google.colab import drive\n",
-    "drive.mount('/content/drive')\n",
-    "\n",
-    "data_set_widget2=widgets.Dropdown(\n",
-    "    options=list(data_set_dict.keys()),\n",
-    "    description='Data Sets:',\n",
+    "options = []\n",
+    "dic={\"Air\":[have_air_viol,have_air_insp,have_air_enf, have_air_comp, have_air_formal], \n",
+    "      \"Water\": [have_water_history, have_water_insp, have_water_enf],\n",
+    "      \"RCRA\": [have_rcra_viol, have_rcra_insp, have_rcra_enf]\n",
+    "     }\n",
+    "for opt in dic:\n",
+    "    print(dic[opt])\n",
+    "    if True in dic[opt]:\n",
+    "        options.append(opt)\n",
+    "print(options)\n",
+    "program_widget=widgets.Dropdown(\n",
+    "    options=options,\n",
+    "    description='Programs:',\n",
     "    disabled=False,\n",
     ")\n",
-    "display(data_set_widget2)"
+    "display(program_widget)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chart_title = program_widget.value\n",
```

Comment:
this code block errors out:

```
['Water QNCR History']
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-15-9f263c9737f7> in <module>
     14         d["Water Inspections"] = d["Water Inspections"].groupby(pd.to_datetime(d["Water Inspections"]['ACTUAL_END_DATE'], format="%m/%d/%Y").dt.to_period("Y"))[["REGISTRY_ID"]].count()
     15     # More spreadsheet prepping needed here
---> 16     program = pd.concat([d[k[0]], d[k[1]]], axis=1, sort=False) # Do for all spreadsheets in k...
     17     #print(program)
     18 elif (program_widget.value == "Air"):

IndexError: list index out of range
```

Reply:
Yea, this was specific to an example I was working with. Will fix.

Reply:
fixed

Reply:
A note: we should merge #5 so that @danpascucci has full access to the code. But some of the code blocks, like this one, will then need to be removed, as we're not going to be able to get them to a ready state before Thursday. However, we will want them for future iterations, so I'll tease them out on a separate branch when the time comes...